### ISSUE TITLE
[REVIEW]: Cacatoo: building, exploring, and sharing spatially structured models of biological systems

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3948]
**Submitting author:** @bramvandijk88 (<a href="http://orcid.org/0000-0002-6330-6934">Bram van Dijk</a>)
**Repository:** <a href="https://github.com/bramvandijk88/cacatoo" target ="_blank">https://github.com/bramvandijk88/cacatoo</a>
**Version:** 1.0.3
**Editor:** @Bisaloo
**Reviewer:** @emilydolson, @TimKam
**Archive:** <a href="https://doi.org/10.5281/zenodo.5918404" target="_blank">10.5281/zenodo.5918404</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a/status.svg)](https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a"><img src="https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a/status.svg)](https://joss.theoj.org/papers/91fee0cc4c10d0d7696b78819072317a)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/bramvandijk88/cacatoo" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@emilydolson & @TimKam, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Bisaloo know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @emilydolson
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/bramvandijk88/cacatoo">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bramvandijk88/cacatoo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@bramvandijk88) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @TimKam
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/bramvandijk88/cacatoo">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bramvandijk88/cacatoo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@bramvandijk88) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @emilydolson, @TimKam it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1093/bioinformatics/btt772 is OK
- 10.7554/eLife.61288 is OK
- 10.1101/2021.05.29.446280 is OK
- 10.1016/j.zool.2019.02.007 is OK
- 10.1098/rstb.2019.0248 is OK
- 10.1101/053405 is OK
- 10.3389/fmicb.2015.00728 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.27 s (428.6 files/s, 285181.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            67          22318            315          19027
SVG                              9              0              0          16611
JavaScript                      27           1465           3089           9484
JSON                             1              0              0           1794
CSS                              6            199             68           1230
TeX                              1              8              0             75
Markdown                         2             32              0             64
Bourne Shell                     1             17              9             54
-------------------------------------------------------------------------------
SUM:                           114          24039           3481          48339
-------------------------------------------------------------------------------


Statistical information for the repository '41d513f16f9bdba7c5f10c3d' was
gathered on 2021/11/23.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Bram van Dijk                   24          3508           1172            0.47
bramvandijk88                  153        490986         472663           96.80
jeroenmeijer                     2            26             24            0.01
root                             4         24015           3150            2.73

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
bramvandijk88             16747            3.4          3.4               23.08
jeroenmeijer                  4           15.4          1.6                0.00
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03948/joss.03948/10.21105.joss.03948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03948/joss.03948/10.21105.joss.03948.pdf) :page_facing_up: :point_left:

**Comment by Bisaloo** [comment]
:wave: :wave: :wave: @bramvandijk88 @emilydolson @TimKam this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#3948` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 4-6 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@bisaloo) if you have any questions/concerns.

**Comment by emilydolson** [comment]
@Bisaloo should I be able to edit this issue to check things off? I don't seem to be able to. I suspect this has to do with me not having accepted the invitation to collaborate on this repository before it expired.

**Comment by Bisaloo** [comment]
@emilydolson, yes, you should be able to edit the checklist once you have accepted the invitation. I'll re-invite you :relaxed:

**Comment by Bisaloo** [comment]
@whedon re-invite @emilydolson as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@emilydolson please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by whedon** [comment]
:wave: @TimKam, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @emilydolson, please update us on how your review is going (this is an automated reminder).

**Comment by bramvandijk88** [comment]
Heya,

Just checking in after the holidays, is this review still in progress? I'm not at all in a hurry, but just wanted to make sure it wasn't forgotten.

**Comment by emilydolson** [comment]
Yes! Apologies for being slow - my university switched teaching modalities and I've been scrambling to adjust my course.

**Comment by bramvandijk88** [comment]
> Yes! Apologies for being slow - my university switched teaching modalities and I've been scrambling to adjust my course.

No need to apologise, there have been holidays + a new year of teaching, so nobody should expect to stay within standardised deadlines. Thanks for your time! :)

**Comment by emilydolson** [comment]
Okay, done with my initial review! Really cool project! @bramvandijk88 already added the contributing guidelines, so in my mind the only two issues currently blocking acceptance are:

- [Improved unit tests](https://github.com/bramvandijk88/cacatoo/issues/14)
- [A little more discussion of the state of the field](https://github.com/bramvandijk88/cacatoo/issues/16)

**Comment by Bisaloo** [comment]
Thank you for your very insighful comments @emilydolson and thanks @bramvandijk88 for sending a reminder (Sorry for my lack of activity, I'm still on leave without reliable internet access until next week)

**Comment by emilydolson** [comment]
I'm satisfied that all items on the review checklist are complete!

**Comment by Bisaloo** [comment]
Thanks @emilydolson!

@TimKam, could you give us an update for your review of this library+paper please? Please let me know if you need anything.

**Comment by TimKam** [comment]
Sorry, I should be able to complete the review within roughly (but perhaps not exactly) one week.

**Comment by TimKam** [comment]
I have completed the review and have created [an issue in the source code repository](https://github.com/bramvandijk88/cacatoo/issues/19) that outlines some shortcomings/improvements, with respect to some issues around installation & instructions, automated tests, community guidelines, and the software paper (minor comments).
Conceptually/w.r.t. scholarly effort, I assess the paper as reasonable (from an engineering point of view, I do not have any expertise in ecology and evolution).

**Comment by TimKam** [comment]
@bramvandijk88 responded rather promptly and we have concluded the revisions and discussions regarding my review. I have now completed the review checklist.

**Comment by bramvandijk88** [comment]
@TimKam not sure if this is intended, but one of your checkboxes is still unticked.

**Comment by TimKam** [comment]
Thx, now it's ticked!

**Comment by Bisaloo** [comment]
Great! Thank you again @emilydolson and @TimKam for your valuable comments!

@bramvandijk88, I'll now do a last review for typos. I encourage you to do the same because there is no proofing after acceptance at JOSS. Once this is done, we can take the final steps to accept this submission.

**Comment by Bisaloo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03948/joss.03948/10.21105.joss.03948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03948/joss.03948/10.21105.joss.03948.pdf) :page_facing_up: :point_left:

**Comment by Bisaloo** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1093/bioinformatics/btt772 is OK
- 10.7554/eLife.61288 is OK
- 10.1101/2021.05.29.446280 is OK
- 10.1016/j.zool.2019.02.007 is OK
- 10.1098/rstb.2019.0248 is OK
- 10.1101/053405 is OK
- 10.3389/fmicb.2015.00728 is OK

MISSING DOIs

- 10.1098/rstb.2020.0460 may be a valid DOI for title: Introduction: the secret lives of microbial mobile genetic elements

INVALID DOIs

- None
```

**Comment by Bisaloo** [comment]
Shouldn't https://github.com/bramvandijk88/cacatoo/blob/65b6e7a2daac8175d65a6052b4f2f6d3652d08b7/paper/paper.bib#L1-L10 be `@article` instead of `@misc` anyways?

**Comment by bramvandijk88** [comment]
I've added the DOI of "Introduction: the secret lives of Mobile genetic Elements".

Google indexed that citation as @misc, probably because it's not an article in the strict sense, it's just an overview of the content in the special issue.

**Comment by bramvandijk88** [comment]
I've checked the proof for typos (caught a few!) Please let me know what happens next.

**Comment by Bisaloo** [comment]
At this point could you:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by bramvandijk88** [comment]
> At this point could you:
> 
> * [x]  Make a tagged release of your software, and list the version tag of the archived version here.
> * [x]  Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
> * [x]  Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
> * [x]  Please list the DOI of the archived version here.

Done. The archived Zenodo DOI is 10.5281/zenodo.5918404

**Comment by Bisaloo** [comment]
@whedon set 1.0.1 as version

**Comment by whedon** [comment]
OK. 1.0.1 is the version.

**Comment by Bisaloo** [comment]
@bramvandijk88, there is a mismatch in the version listed in Zenodo and the actual tag in GitHub. If you can easily fix it, it would be great. But if it's too difficult, then we can ignore it and proceed.

Additionally, could you please edit the license on Zenodo. It's listed as 'Other' when it should be 'GPL-3'?

**Comment by bramvandijk88** [comment]
What mismatch is there? They are both 1.0.1 as far as I can tell. 

I've changed the licence to "GPL-3" in the zenodo.json file, but it only updates this upon republishing the package. Zenodo complains when I do that without bumping the version, but I actually don't want to bump the version without having any changes in the code. Is there a way around this?

**Comment by Bisaloo** [comment]
Here is what I see. And the link to 1.0.3 on GitHub doesn't work because this version doesn't exist.

![image](https://user-images.githubusercontent.com/10783929/151954428-00e93aac-e77f-4c3c-9be2-5bf8df0530bd.png)

**Comment by bramvandijk88** [comment]
Not sure what's going on here. V1.0.3 was the result of me testing the metadata, for which I was forced to publish (or Zenodo wouldn't reindex them). 

I'm not sure how to resolve this. There's no way to edit these things after publication as far as I can tell, but the DOI-link on the github page itself should get you to version 1.0.1.

**Comment by Bisaloo** [comment]
Okay, let's go with this then. It's not super important.

**Comment by Bisaloo** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by Bisaloo** [comment]
@whedon set 10.5281/zenodo.5918404 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5918404" target="_blank">10.5281/zenodo.5918404</a> is the archive.

**Comment by Bisaloo** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1098/rstb.2020.0460 is OK
- 10.1093/bioinformatics/btt772 is OK
- 10.7554/eLife.61288 is OK
- 10.1101/2021.05.29.446280 is OK
- 10.1016/j.zool.2019.02.007 is OK
- 10.1098/rstb.2019.0248 is OK
- 10.1101/053405 is OK
- 10.3389/fmicb.2015.00728 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2921

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2921, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 1.0.3 as version

**Comment by whedon** [comment]
OK. 1.0.3 is the version.

**Comment by Kevin-Mattheus-Moerman** [comment]
@bramvandijk88 @Bisaloo I have read the paper and inspected the archive. I only had to update the version since the archived version was 1.0.3. All seems in order now to proceed to acceptance.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1489881908901203971) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2930
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03948](https://doi.org/10.21105/joss.03948)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
Congratulations @bramvandijk88 on your publication! :tada: 

Thank you @emilydolson and @TimKam for your review efforts!! 

Thanks @Bisaloo for editing this.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03948/status.svg)](https://doi.org/10.21105/joss.03948)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03948">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03948/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03948/status.svg
   :target: https://doi.org/10.21105/joss.03948
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03948/status.svg)](https://doi.org/10.21105/joss.03948)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
